components/ledger/pkg/ledger/ledger.go

```diff
@@ -41,9 +50,14 @@ func (l *Ledger) GetLedgerStore() storage.LedgerStore {
 }
 
 func (l *Ledger) CreateTransaction(ctx context.Context, dryRun bool, script core.RunScript) (*core.ExpandedTransaction, error) {
-	return l.runner.Execute(ctx, script, dryRun, func(expandedTx core.ExpandedTransaction, accountMetadata map[string]core.Metadata) core.Log {
+	tx, log, err := l.runner.Execute(ctx, script, dryRun, func(expandedTx core.ExpandedTransaction, accountMetadata map[string]core.Metadata) core.Log {
 		return core.NewTransactionLog(expandedTx.Transaction, accountMetadata)
 	})
+	if err == nil {
+		err = l.queryWorker.QueueLog(ctx, log, l.store)
```

Comment:
I think QueueLog() should not return any error.
In this case, even if we return an error, the log will be written on disk.

Reply:
Since we have a way to compute it again with the cache, i'm not against doing nothing on this case. Do you think just removing the error will be fine on your side ?